include/openthread/srp_client.h

```diff
@@ -0,0 +1,501 @@
+ *
+ *  - The SRP client is started - `otSrpClientStart()` is called.
+ *  - Host name is set - `otSrpClientSetHostName()` is called.
+ *  - At least one host IPv6 address is set - `otSrpClientSetHostName()` is called.
+ *  - At least one service is added - `otSrpClientAddService()` is called.
+ *
+ * It does not matter in which order these functions are called. When all conditions are met, the SRP client will
+ * wait for a short delay before preparing an "SRP Update" message and sending it to server. This delay allows for user
+ * to add multiple services and/or IPv6 addresses before the first SRP Update message is sent (ensuring a single SRP
+ * Update is sent containing all the info). The config `OPENTHREAD_CONFIG_SRP_CLIENT_UPDATE_TX_DELAY` specifies the
+ * delay interval.
+ *
+ * @param[in] aInstance        A pointer to the OpenThread instance.
+ * @param[in] aServerSockAddr  The socket address (IPv6 address and port number) of the SRP server.
+ * @param[in] aCallback        The callback which is used to notify events and changes. Can be NULL if not needed.
+ * @param[in] aContext         An arbitrary context used with @p aCallback.
+ *
+ * @retval OT_ERROR_NONE       SRP client operation started successfully or it is already running with same server
+ *                             socket address and callback.
+ * @retval OT_ERROR_BUSY       SRP client is busy running with a different socket address and/or callback.
+ * @retval OT_ERROR_FAILED     Failed to open/connect the client's UDP socket.
+ *
+ */
+otError otSrpClientStart(otInstance *        aInstance,
+                         const otSockAddr *  aServerSockAddr,
+                         otSrpClientCallback aCallback,
+                         void *              aContext);
+
+/**
+ * This function stops the SRP client operation.
+ *
+ * This function stops any further interactions with the SRP server. Note that it does not remove or clear host info
+ * and/or list of services. It marks all services to be added/removed again once the client is (re)started.
+ *
+ * @param[in] aInstance       A pointer to the OpenThread instance.
+ *
+ */
+void otSrpClientStop(otInstance *aInstance);
+
+/**
+ * This function gets the lease interval used in SRP update requests.
+ *
+ * Note that this is the lease duration requested by the SRP client. The server may choose to accept a different lease
+ * interval.
+ *
+ * @param[in] aInstance        A pointer to the OpenThread instance.
+ *
+ * @returns The lease interval (in seconds).
+ *
+ */
+uint32_t otSrpClientGetLeaseInterval(otInstance *aInstance);
+
+/**
+ * This function sets the lease interval used in SRP update requests.
+ *
+ * Changing the lease interval does not impact the accepted lease interval of already registered services/host-info.
+ * It only affects any future SRP update messages (i.e., adding new services and/or refreshes of the existing services).
```

Comment:
Do we allow setting lease interval to `0`. What happens when `0` is set? Will the host and services be removed?

Reply:
I think there should only be one way to remove. We should probably prohibit setting 0.

Reply:
I agree. It's better to have one way to remove. 
If it zero, then we use the default value. Will add/update comments to make this clear.